### PR TITLE
Inconsistent unread messages count updating #873

### PR DESCRIPTION
**Pull request by novokrest**
@jcbrand Hi! Because of PR turned out to be quite large, I decide to create separate issue (just for history purposes) for which I made these changes.

### CODE REVIEW

File: tests/utils.js [hunk 1]
```diff
@@ -216,6 +217,17 @@
...
+    utils.createChatMessage = function (_converse, sender_jid, message) {
+        return $msg({
+                   from: sender_jid,
+                   to: _converse.connection.jid,
+                   type: 'chat',
+                   id: (new Date()).getTime()
+               })
+               .c('body').t(message).up()
+               .c('active', {'xmlns': 'http://jabber.org/protocol/chatstates'}).tree();
+    }
```
Comment (jcbrand) [inline tests/utils.js]: For the 'xmlns' value, please use the `Strophe.NS.CHATSTATES`.

File: src/converse-core.js [hunk 2]
```diff
@@ -605,6 +600,7 @@
                 _converse.clearMsgCounter();
             }
             _converse.windowState = state;
+            _converse.emit('windowStateChanged', {state: state})
```
Comment (jcbrand) [inline src/converse-core.js]: Please document any new events you create here: https://github.com/jcbrand/converse.js/blob/master/docs/source/events.rst

File: src/converse-minimize.js [hunk 3]
```diff
@@ -312,15 +320,7 @@
...
-                    this.model.messages.on('add', function (m) {
-                        if (m.get('message')) {
-                            this.updateUnreadMessagesCounter();
-                        }
-                    }, this);
-                    this.model.on('change:minimized', this.clearUnreadMessagesCounter, this);
-                    // OTR stuff, doesn't require this module to depend on OTR.
-                    this.model.on('showReceivedOTRMessage', this.updateUnreadMessagesCounter, this);
-                    this.model.on('showSentOTRMessage', this.updateUnreadMessagesCounter, this);
+                    this.model.on('change:num_unread', this.updateUnreadMessagesCounter, this);
```
Comment (jcbrand) [inline src/converse-minimize.js]: Looks like `updateUnreadMessagesCounter` can be removed and `this.render` used here.

**Comment by jcbrand** [comment]
Great work @novokrest, thank you! I made three minor comments to your PR.

File: tests/utils.js (hunk 1 above)
Comment (novokrest) [inline tests/utils.js]: @jcbrand Sure!

File: src/converse-core.js (hunk 2 above)
Comment (novokrest) [inline src/converse-core.js]: Added short description

File: src/converse-minimize.js (hunk 3 above)
Comment (novokrest) [inline src/converse-minimize.js]: Yes, exactly

**Pull request merged**